src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -0,0 +1,239 @@
+@doc raw"""
+    CompressibleNavierStokes2D(gamma,
+                               Re,
+                               Pr,
+                               Ma_inf,
+                               kappa,
+                               equations)
+
+`CompressibleNavierStokes2D` represents the diffusion (i.e. parabolic) terms applied
+to mass, momenta, and total energy together with the advective from
+the `CompressibleEulerEquations2D`.
+
+gamma: adiabatic constant,
+Re: Reynolds number,
+Pr: Prandtl number,
+Ma_inf: free-stream Mach number
+kappa: thermal diffusivity for Fick's law
+
+For the particular scaling the vicosity is set internally to be μ = 1/Re.
+Further, the nondimensionalization takes the density-temperature-sound speed as
+the principle quantities such that
+rho_inf = 1.0
+T_ref = 1.0
+c_inf = 1.0
+p_inf = 1.0 / gamma
+u_inf = Ma_inf
+R = 1.0 / gamma
+
+Other normalization strategies exist, see the reference below for details.
+- Marc Montagnac (2013)
+  Variable Normalization (nondimensionalization and scaling) for Navier-Stokes
+  equations: a practical guide
+  [CERFACS Technical report](https://www.cerfacs.fr/~montagna/TR-CFD-13-77.pdf)
+The scaling used herein is Section 4.5 of the reference.
+
+In two spatial dimensions we require gradients for three quantities, e.g.,
+primitive quantities
+  grad(u), grad(v), and grad(T)
+or the entropy variables
+  grad(w_2), grad(w_3), grad(w_4)
+where
+  w_2 = rho v_1 / p, w_3 = rho v_2 / p, w_4 = -rho / p
+"""
+# TODO:
+# 1) For now I save gamma and inv(gamma-1) again, but we could potentially reuse them from
+#    the Euler equations
+# 2) Add more here and probably some equations
+struct CompressibleNavierStokes2D{RealT<:Real, E} <: AbstractCompressibleNavierStokesEquations{2, 3}
+  gamma::RealT               # ratio of specific heats
+  inv_gamma_minus_one::RealT # = inv(gamma - 1); can be used to write slow divisions as fast multiplications
+  Re::RealT                  # Reynolds number
+  Pr::RealT                  # Prandtl number
+  Ma_inf::RealT              # free-stream Mach number
+  kappa::RealT               # thermal diffusivity for Fick's law
+
+  p_inf::RealT               # free-stream pressure
+  u_inf::RealT               # free-stream velocity
+  R::RealT                   # gas constant (depends on nondimensional scaling!)
+
+  equations::E               # CompressibleEulerEquations2D
+end
+
+function CompressibleNavierStokes2D(gamma, Reynolds, Prandtl, Mach_freestream, kappa, equations)
+  γ, inv_gamma_minus_one = promote(gamma, inv(gamma - 1))
```

Comment:
Do you mean
```julia
p_inf = promte(1.0 / γ)
```

Reply:
I was thinking about promoting all arguments, just in case, i.e., something like
```julia
γ, inv_gamma_minus_one, Re, Pr, Ma, κ = promote(gamma, inv(gamma - 1), Reynolds, Prandtl, Mach_freestream, kappa)
```
This will ensure things as they should if someone mixes different argument types (some `Int`s, some `Float32`s, some `Float64`s etc.).

Reply:
Done